src/coreclr/src/jit/importer.cpp

```diff
@@ -5941,6 +5941,28 @@ int Compiler::impBoxPatternMatch(CORINFO_RESOLVED_TOKEN* pResolvedToken, const B
                                 }
                             }
                         }
+                        break;
+
+                    // box + isinst + unbox.any
+                    case CEE_UNBOX_ANY:
+                        if ((nextCodeAddr + 5) <= codeEndp)
+                        {
+                            CORINFO_RESOLVED_TOKEN unboxResolvedToken = {};
+                            impResolveToken(codeAddr + 1, &unboxResolvedToken, CORINFO_TOKENKIND_Class);
```

Comment:
For the unbox token, this should be `nextCodeAddr + 1`, right?

Shouldn't we be verifying that all 3 tokens in this sequence (box, isinst, unbox.any) must compare equal?

Reply:
Good point!